### PR TITLE
feat: upgrade jax version to 0.8.0

### PR DESCRIPTION
**Pull request by jimoosciuc**
### Movitation
Upgrade jax version to 0.8.0.  More information refers to https://github.com/jax-ml/jax/releases/tag/jax-v0.8.0

### Accuracy
```
evalscope eval  --model Qwen/Qwen3-8B --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 32
```
image

### Benchmark
```
python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 16 --random-input 1024 --random-output 1024 --random-range-ratio 1 --warmup-requests 0 --max-concurrency 16
```
image

### CODE REVIEW

**Comment by jimoosciuc** [comment]
> @jimoosciuc We need to upgrade the Python version > 3.12 for fixing CI.

DONE

**Comment by pathfinder-pf** [comment]
LGTM
